newbank/server/authentication/Authenticator.java

```diff
@@ -0,0 +1,67 @@
+package newbank.server.authentication;
+
+import de.taimos.totp.TOTP;
+import newbank.server.customers.CustomerID;
+import org.apache.commons.codec.binary.Base32;
+import org.apache.commons.codec.binary.Hex;
+import java.security.SecureRandom;
+import java.util.HashMap;
+
+public class Authenticator {
+    public static Authenticator theInstance;
+    private HashMap<CustomerID, String> authenticatorKeys;
+    public String code;
+
+    private Authenticator() {
+        authenticatorKeys = new HashMap<>();
+    }
+
+    public static Authenticator getInstance() {
+        if (theInstance == null) {
+            theInstance = new Authenticator();
+        }
+
+        return theInstance;
+    }
+
+    public String generateSecretKey() {
```

Comment:
Just curious, what is the key generated here used for? Does it have something to do with linking the authenticator to the account?

Reply:
Answered in the SetupAutheticator command.